proposals/NNNN-Deprecated-interfaces-markup.md

```diff
@@ -0,0 +1,133 @@
+#else 
+#ifdef __GNUC__
+#define DEPRECATED __attribute__((deprecated))
+#define DEPRECATED_CLASS __attribute__((deprecated))
+#elif defined(_MSC_VER)
+#define DEPRECATED __declspec(deprecated)
+#else
+#pragma message("WARNING: You need to implement DEPRECATED for this compiler")
+#define DEPRECATED
+#endif
+#endif
+```
+
+It will generate compile time warning in with c++98 or c++11 or c++14.
+
+### Usage : 
+
+#### C++ Code :
+```cpp
+DEPRECATED void bar() {}
+int main() {
+  bar();
+}
+```
+
+#### Compilers output : 
+
+##### G++ :
+```bash
+$ g++ --std=c++11 -Werror -Wno-error=deprecated-declarations ./main.cc                            
+./main.cc: In function ‘int main()’:
+./main.cc:85:5: warning: ‘void bar()’ is deprecated (declared at ./main.cc:19) [-Wdeprecated-declarations]
+     bar();
+     ^
+./main.cc:85:9: warning: ‘void bar()’ is deprecated (declared at ./main.cc:19) [-Wdeprecated-declarations]
+     bar();
+
+$  g++ --std=c++98 -Werror -Wno-error=deprecated-declarations ./main.cc 
+./main.cc: In function ‘int main()’:
+./main.cc:85:5: warning: ‘void bar()’ is deprecated (declared at ./main.cc:19) [-Wdeprecated-declarations]
+     bar();
+     ^
+./main.cc:85:9: warning: ‘void bar()’ is deprecated (declared at ./main.cc:19) [-Wdeprecated-declarations]
+     bar();
+         ^
+```
+
+##### Clang : 
+```bash
+$ clang --std=c++98 -Werror -Wno-error=deprecated-declarations ./main.cc  
+./main.cc:23:5: warning: 'bar' is deprecated [-Wdeprecated-declarations]
+    bar();
+    ^
+./main.cc:19:17: note: 'bar' has been explicitly marked deprecated here
+DEPRECATED void bar() {}
+                ^
+1 warning generated.
+
+$ clang --std=c++11 -Werror -Wno-error=deprecated-declarations ./main.cc  
+./main.cc:23:5: warning: 'bar' is deprecated [-Wdeprecated-declarations]
+    bar();
+    ^
+./main.cc:19:17: note: 'bar' has been explicitly marked deprecated here
+DEPRECATED void bar() {}
+                ^
+1 warning generated.
+```
+
+Full usage example is available https://github.com/alexkutsan/RnD_fun/blob/master/deprecated/main.cc 
+
+Also doxygen `@deprecated` command should be used for documentation:
+```cpp
+/**
+ * @brief The Y class containt one actual and one deprecated method
+ */
+class Y {
+
+public:
+
+    /**
+     * @brief foo not deprecated function of not deprecated class
+     */
+    void foo(){}
+
+    /**
+     * @brief bar deprecated function of not deprecated class
+     * @deprecated
+     */
+    DEPRECATED int bar(){}
+};
+```
+
+![Doxygen result](http://i.imgur.com/dFUDtSu.png)                                                                                                                                                                                                                                                    
+
+## Impact on existing code
+No changes in behaviour should be introduced. 
+In some SDL core methods should be marked as deprecated and using this methods will cause compile time warning
```

Comment:
Can you please clarify this sentence?  Should it read "In some SDL core methods, the methods should be marked as deprecated and using these methods will cause a compile time warning." ?

Reply:
@theresalech there are already some amount of deprecated methods (marked with simle comments). 
But I can't say that this methods not used. If methods used, it will cause compile time warnings ( by design)